### PR TITLE
void creating strings when not needed.

### PR DESCRIPTION
**Pull request by baldersheim**
Do not depend on toString(), use toUtf8 instead.

@vekterli and @bratseth PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@hmusum PR too

**Review by bratseth — APPROVED** [review]

File: component/src/main/java/com/yahoo/component/Version.java [hunk 1]
```diff
@@ -270,10 +276,14 @@ public String toFullString() {
      * The string representation of a Version specified here is a part of the API and will never change.
      */
     @Override
-    public String toString() { return stringValue; }
+    public String toString() { return toStringValue(); }
+
+    public Utf8Array toUtf8() {
+        return utf8 != null ? utf8 : new Utf8String(toString());
```
Comment (vekterli) [inline component/src/main/java/com/yahoo/component/Version.java]: Can `utf8` ever be `null` here?

**Review by vekterli — APPROVED** [review]
:+1: one question about possible simplification

**Review by hmusum — APPROVED** [review]

File: component/src/main/java/com/yahoo/component/Version.java (hunk 1 above)
Comment (baldersheim) [reply component/src/main/java/com/yahoo/component/Version.java]: Good point, not anymore.

**Pull request merged**